rover-apps/common/include/WatchdogModule.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "Module.h"
+#include "mbed.h"
+
+class WatchdogModule final : public Module {
+ public:
+  /* Initiates the watchdog with a countdown
+   * @param countdown - max timeout of the watchdog
+   * */
+  WatchdogModule(std::chrono::milliseconds countdown = 5000ms);
+
+  /* Logs the last reset due to the watchdog
+   * */
+  void Watchdog_logLastResetReason(void);
+
+  /* Periodic function to kick the watchdog and restart its timer every 1s
+   * */
+  void periodic_10s(void) override {}
+  void periodic_1s(void) override;
+  void periodic_100ms(void) override {}
+  void periodic_10ms(void) override {}
+  void periodic_1ms(void) override {}
+
+ private:
+  std::chrono::milliseconds countdown_ms; /*max timeout of the watchdog*/
```

Comment:
This doesn't need to be stored in a member var. I suggest making a static constant `WATCHDOG_DEFAULT_COUNTDOWN = 5000ms`, and use this as the default value passed into the constructor. Since you're petting the dog every 1s, it would be a good idea to make the countdown longer than 1s.